compiler-plugin/src/main/java/io/ballerina/lib/ai/np/compilerplugin/provider/PromptGenerator.java

```diff
@@ -0,0 +1,242 @@
+package io.ballerina.lib.ai.np.compilerplugin.provider;
+
+import com.google.gson.JsonArray;
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ConstantSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.syntax.tree.InterpolationNode;
+import io.ballerina.compiler.syntax.tree.LiteralValueToken;
+import io.ballerina.compiler.syntax.tree.NaturalExpressionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+
+import java.io.IOException;
+
+import static io.ballerina.lib.ai.np.compilerplugin.provider.ProviderUtils.CONTENT;
+import static io.ballerina.lib.ai.np.compilerplugin.provider.ProviderUtils.FILE_PATH;
+import static io.ballerina.lib.ai.np.compilerplugin.provider.ProviderUtils.retrieveLangLibs;
+
+public class PromptGenerator {
+    private static final String LANG_LIBS_PATH = "/langlibs.json";
+    private static final String LANG_LIBS;
+    static {
+        try {
+            LANG_LIBS = retrieveLangLibs(LANG_LIBS_PATH);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static String getSystemPromptForFunction() {
+        return """
+            Langlibs
+            <langlibs>
+               %s
+            </langlibs>
+            
+            You are a Ballerina code generation assistant specialized in replacing external functions 
+            annotated with @natural:code. Your task is to generate function implementations that 
+            satisfy natural language requirements.
+            
+            Follow these guidelines when generating code:
+            
+            1. Function Implementation:
+               - Generate ONLY the function implementation with the exact same signature as the original
+               - Include ONLY the required `ballerina` and `ballerinax` organization imports
+               - Use only function parameters passed to the function and do not use configurable 
+                 variables or module-level variables from the program
+            
+            2. Code Standards:
+               - Use two words camel case identifiers (variables, parameters, field names)
+               - Use dot notation for normal functions, -> for remote/resource functions
+               - Use named arguments when providing values to parameters (e.g., .get(key="value"))
+               - Mention types EXPLICITLY in variable declarations and foreach statements
+               - Use // for single line comments, avoid long comments
+               - Ensure proper error handling and type checking
```

Comment:
```suggestion
               - Ensure proper error handling
```

We shouldn't mix up code validity and guidelines IMO.